### PR TITLE
Support Cookie CHIPS

### PR DESCRIPTION
**Pull request by pgjones**
CHIPS, Cookies Having Independent Partitioned State, allows for cookies to be opted into partitioned storage which is especially useful for thrid party cookies.

Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: src/werkzeug/http.py [hunk 1]
```diff
@@ -1252,9 +1253,13 @@ def dump_cookie(
         <cookie_>`_. Set to 0 to disable this check.
     :param samesite: Limits the scope of the cookie such that it will
         only be attached to requests if those requests are same-site.
+    :param oartitioned:
```
Comment (davidism) [inline src/werkzeug/http.py:1256]: typo

File: CHANGES.rst [hunk 2]
```diff
@@ -1,5 +1,13 @@
 .. currentmodule:: werkzeug
 
+Version 3.1.0
+-------------
+
+Unreleased
+
+-  Support Cookie CHIPS (Partitioned Cookies). :pr:`2855`
```
Comment (davidism) [inline CHANGES.rst:8]: Can link to the issue https://github.com/pallets/werkzeug/issues/2797

File: src/werkzeug/http.py [hunk 3]
```diff
@@ -1298,6 +1303,9 @@ def dump_cookie(
         if samesite not in {"Strict", "Lax", "None"}:
             raise ValueError("SameSite must be 'Strict', 'Lax', or 'None'.")
 
+    if partitioned and not secure:
```
Comment (davidism) [inline src/werkzeug/http.py:1306]: Would it make sense to automatically force `secure=True` for `partitioned=True`?

File: src/werkzeug/http.py (hunk 1 above)
Comment (pgjones) [reply src/werkzeug/http.py]: Ta

File: CHANGES.rst (hunk 2 above)
Comment (pgjones) [reply CHANGES.rst]: Yep

File: src/werkzeug/http.py (hunk 3 above)
Comment (pgjones) [reply src/werkzeug/http.py:1306]: I'm not sure, would it be confusing if `dump_cookie(..., partitioned=True, secure=False)` worked even though it shouldn't?

Comment (davidism) [reply src/werkzeug/http.py:1306]: We could make `secure=None` by default to detect if `False` is passed explicitly, but I don't really think it's worth it. If the docs for the `partitioned` argument say "This forces `secure=True`." I think that's enough.

Comment (pgjones) [reply src/werkzeug/http.py]: Ok, I've changed.

**Comment by hmoffatt** [comment]
How do you use this without `Response.set_cookie` also being modified to add the new parameter?

**Pull request merged**